test/distributed/org/apache/cassandra/fuzz/topology/AccordTopologyMixupTest.java

```diff
@@ -0,0 +1,397 @@
+
+            SharedState.listeners.add(this);
+        }
+
+        @Override
+        protected void onConfigure(IInstanceConfig c)
+        {
+            c.set("accord.shard_count", 1)
+             .set("paxos_variant", Config.PaxosVariant.v2.name());
+        }
+
+        @Override
+        public void debugTxn(@Nullable Node.Id exclude, String type, TxnId txnId)
+        {
+            onError.add(new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    // this runs in the main thread, so is actually thread safe
+                    int[] up = topologyHistory.up();
+                    logger.error("{} failed with txn id {}; global debug summary:\n{}", type, txnId, ClusterUtils.queryTxnStateAsString(cluster, txnId, up));
+                    onError.remove(this);
+                }
+            });
+        }
+
+        @Override
+        public void close() throws Exception
+        {
+            for (Runnable r : onError)
+            {
+                try
+                {
+                    r.run();
+                }
+                catch (Throwable t)
+                {
+                    // TODO (correctness): how to handle?
+                    logger.error("Unhandled error in onError listeners", t);
+                }
+            }
+            onError.clear();
+            SharedState.listeners.remove(this);
+            super.close();
+        }
+    }
+
+    @Shared
+    public static class SharedState
+    {
+        public interface Listener
+        {
+            void debugTxn(Node.Id node, String type, TxnId txnId);
+        }
+
+        public static final CopyOnWriteArrayList<Listener> listeners = new CopyOnWriteArrayList<>();
+
+        public static void debugTxn(@Nullable Integer node, String type, String id)
+        {
+            Node.Id nodeId = node == null ? null : new Node.Id(node);
+            TxnId txnId = TxnId.parse(id);
+            listeners.forEach(l -> l.debugTxn(nodeId, type, txnId));
+        }
+    }
+
+    @Isolated
+    public static class InterceptAgent extends AccordAgent
+    {
+        @Override
+        public void onFailedBarrier(TxnId id, Seekables<?, ?> keysOrRanges, Throwable cause)
+        {
+            if (cause instanceof Timeout || cause instanceof Preempted)
+            {
+                SharedState.debugTxn(null, "Repair Barrier", id.toString());
+            }
+        }
+
+        @Override
+        public void onFailedBootstrap(String phase, Ranges ranges, Runnable retry, Throwable failure)
+        {
+            if (failure instanceof Exhausted)
+            {
+                Exhausted e = (Exhausted) failure;
+                SharedState.debugTxn(self.id, "Bootstrap#" + phase, e.txnId().toString());
+            }
+            super.onFailedBootstrap(phase, ranges, retry, failure);
+        }
+    }
+
+    @Isolated
+    public static class Json
```

Comment:
think this is dead code now... with the new vtable to inspect txn we don't need this...